### PR TITLE
Autodeploy SDK & Watch packages

### PR DESCRIPTION
**Pull request by SaraVieira**
Autodeploy to NPM:

- Added a new workflow for SDK that is only triggered if `react-sdk` directory has changed.
- Only newer versions are published to NPM (checked by [this](https://github.com/marketplace/actions/npm-publish)).
- Other changes are based on this [PR](https://github.com/webstudio-is/webstudio-designer/pull/163/files).

Example [run](https://github.com/andrasbacsai/webstudio-designer/runs/7426398746?check_suite_focus=true).

Watch `react-sdk` while developing designer:

- Needed to modify `main` to point to the build assets (for NPM publish).
- Added `tsup` as dev script for sdk.
- Added `watchPaths` to designer remix configuration to rebuild designer in case SDK is modified.

Todos:
- [x] Add `NPM_TOKEN` to repository @kof 

Done by @andrasbacsai

### CODE REVIEW

**Comment by SaraVieira** [comment]
Action fixed but vercel does not run turbo but it runs inside the apps/designer directory and so it can't find the packages :/

**Comment by SaraVieira** [comment]
Getting this error whern trying to build the SDK in vercel

image

Any idea @kof?

**Comment by andrasbacsai** [comment]
It looks like there is no `-S` option in Vercel's special environment for `/usr/bin/env` [command](https://github.com/webstudio-is/webstudio-designer/blob/main/packages/react-sdk/bin/mdn-data.ts#L1) (lol). I will take a look at this.

**Comment by kof** [comment]
> Action fixed but vercel does not run turbo but it runs inside the apps/designer directory and so it can't find the packages :/

@saravieira yes, it starts at the designer dir, but you can do cd ../../ and then do whatever

**Comment by andrasbacsai** [comment]
Fixed the issue with [this](https://github.com/webstudio-is/webstudio-designer/pull/167/commits/d25c5dd1270797ff80d90905163ddac3ea63710d).

**Comment by kof** [comment]
I added the npm key for development environment, should it be added for production too?

**Comment by andrasbacsai** [comment]
> I added the npm key for development environment, should it be added for production too?

Yes, please. If it does not work, you can add it to the repository secrets.

**Comment by SaraVieira** [comment]
I think the designer build is not waiting for everything to finish :/

**Comment by kof** [comment]
Added an the same npm token to all 3 environments

**Comment by andrasbacsai** [comment]
Hmm, not sure why it does not work. Can we add it as a repository secret as well, just for testing?

File: apps/designer/remix.config.js [hunk 1]
```diff
@@ -17,4 +17,7 @@ module.exports = {
   // assetsBuildDirectory: "public/build",
   serverBuildPath: "api/index.js",
   // publicPath: "/build/",
+  watchPaths: () => {
+    return ["../../packages/**/*"];
+  },
```
Comment (kof) [inline apps/designer/remix.config.js:22]: needed this one now, added in main directly

**Comment by kof** [comment]
@andrasbacsai 
Screenshot 2022-07-21 at 22 06 39

**Review by kof — APPROVED** [review]
is this still blocked?

**Comment by andrasbacsai** [comment]
> is this still blocked?

The npm publish is working now (it failed because the version is not changed, so it does not force publishing it). 

But now we have another problem related to asset-uploader (@SaraVieira).
image

I can merge this as it is unrelated to the changes here.

**Pull request merged**